src/viam/ur/module/ur_arm.cpp

```diff
@@ -796,23 +811,50 @@ void URArm::keep_alive_() {
 void URArm::move_(std::shared_lock<std::shared_mutex> config_rlock, std::list<Eigen::VectorXd> waypoints, const std::string& unix_time) {
     auto our_config_rlock = std::move(config_rlock);
 
+    // get current joint position and add that as starting pose to waypoints
+    VIAM_SDK_LOG(info) << "move: get_joint_positions start " << unix_time;
+    auto curr_joint_pos_deg = get_joint_positions_(our_config_rlock);
```

Comment:
Well, `get_joint_positions_` actually starts out in radians internally, and converts to degrees to return, but then you are converting back. The following might clear it up:

- Rename `get_joint_positions_` to `get_joint_positions_rad_`, and remove the conversion to degrees from it. You might even be able to change the return type to `vector6d` to avoid an allocation.
- Add the conversion from radians to degrees to `get_joint_positions` (the one without a trailing `_`).

Then here you can just call `get_joint_positions_rad_` and you won't need to convert back, and you can just make an Eigen::VectorXd that `Maps` over it too, I think.

Reply:
agreed and changed. Can you take a  look at get_joint_positions?